src/include/btree_cmp_inline.h

```diff
@@ -14,12 +14,193 @@
 #endif
 #endif
 
-#if defined(HAVE_ARM_NEON_INTRIN_H)
-#include <arm_neon.h>
-#endif
-/* 16B alignment */
-#define WT_ALIGNED_16(p) (((uintptr_t)(p)&0x0f) == 0)
 #define WT_VECTOR_SIZE 16 /* chunk size */
+#define WT_COMPARE_SHORT_MAXLEN 16
+
+#ifdef HAVE_X86INTRIN_H
+
+#define WT_VECTOR_MASK 0xFFFF
+/*
+ * __lex_compare_gt_16 --
+ *     Lexicographic comparison routine for data greater than 16 bytes. Returns: < 0 if user_item is
+ *     lexicographically < tree_item, = 0 if user_item is lexicographically = tree_item, > 0 if
+ *     user_item is lexicographically > tree_item. We use the names "user" and "tree" so it's clear
+ *     in the btree code which the application is looking at when we call its comparison function.
+ */
+static WT_INLINE int
+__lex_compare_gt_16(const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp)
+{
+    __m128i res_eq, t, u;
+    int32_t eq_bits;
+    size_t i, final_bytes;
+
+    final_bytes = len - WT_VECTOR_SIZE;
+
+    /*
+     * Compare 16 bytes at a time until we find a difference or run out of 16 byte chunks to
+     * compare.
+     */
+    for (i = 0; i < final_bytes; i += WT_VECTOR_SIZE) {
+        u = _mm_loadu_si128((const __m128i *)(ustartp + i));
+        t = _mm_loadu_si128((const __m128i *)(tstartp + i));
+        res_eq = _mm_cmpeq_epi8(u, t);
+        if ((eq_bits = _mm_movemask_epi8(res_eq)) != WT_VECTOR_MASK)
+            goto final128;
+    }
+
+    /*
+     * Rewind until there is exactly 16 bytes left. We know we started with at least 16, so we are
+     * still in bounds.
+     */
+    i = final_bytes;
+    u = _mm_loadu_si128((const __m128i *)(ustartp + i));
+    t = _mm_loadu_si128((const __m128i *)(tstartp + i));
+    res_eq = _mm_cmpeq_epi8(u, t);
+    eq_bits = _mm_movemask_epi8(res_eq);
+
+    if (eq_bits == WT_VECTOR_MASK)
+        return (lencmp);
+    else {
+final128:
+        /* The initial matching bytes correspond to trailing 1 bits in eq_bits. */
+#ifndef _MSC_VER
+        i += (size_t)__builtin_ctz(~(uint32_t)eq_bits);
+#else
+        unsigned long res;
+        _BitScanForward(&res, ~(uint32_t)eq_bits);
+        i += res;
+#endif
+
+        /* C zero-extends the bytes to 32 bit integers before the calculation. */
+        return ((int)(ustartp[i] - tstartp[i]));
+    }
+}
+#else
```

Comment:
Can you put a comment what this else is for? I had to go up and figure it out, and it's not assisted by indenting.